R/sample.R

```diff
@@ -270,3 +290,18 @@ restart_data <- function(res, model, sampler, runner) {
        sampler = sampler,
        runner = runner)
 }
+
+
+direct_sample_within_domain <- function(model, rng, max_attempts = 100) {
```

Comment:
It's ok for some attempts to fail to be within the domain then?

Reply:
yeah, I'm not sure about this really. I expect that the user might pass in a function that usually works, but we have to try a few times to get it right and we should sample a couple of times until we hit something reasonable.